source/extensions/common/wasm/context.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <memory>
+
+#include "include/proxy-wasm/context.h"
+
+#include "envoy/access_log/access_log.h"
+#include "envoy/buffer/buffer.h"
+#include "envoy/http/filter.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+
+class Wasm;
+
+#include "proxy_wasm_common.h"
+
+using proxy_wasm::BufferInterface;
+using proxy_wasm::MetricType;
+using proxy_wasm::Pairs;
+using proxy_wasm::PairsWithStringValues;
+using proxy_wasm::PluginBase;
+using proxy_wasm::WasmBase;
+using proxy_wasm::WasmBufferType;
+using proxy_wasm::WasmHandle;
+using proxy_wasm::WasmResult;
+using proxy_wasm::WasmVm;
+
+using PluginBaseSharedPtr = std::shared_ptr<PluginBase>;
+
+class Buffer : public proxy_wasm::BufferInterface {
+public:
+  Buffer() {}
+
+  // Add additional wrappped types here.
+  Buffer* set(absl::string_view data) {
+    data_ = data;
+    return this;
+  }
+
+  size_t size() const override { return data_.size(); }
```

Comment:
For clarity please use Envoy style of having a single line '//' comment above each set of override methods saying which class they are overridden from. Same elsewhere in this PR.

Reply:
Done.  PTAL